File: src/ackward/logging/Filter.cpp
#include <ackward/logging/Filter.hpp>

#include <ackward/logging/Module.hpp>

#include <boost/python.hpp>

using namespace boost::python;

namespace ackward { namespace logging
{

Filter::Filter() :
    Object (
        module().attr("Filter")())
{}

Filter::Filter(const std::wstring& name) :
    Object (
        module().attr("Filter")(name))
{}

Filter::Filter(boost::python::object f) :
    Object (f)
{}

// Retained for future use
//
// #include <boost/python/make_function.hpp>
//
// ExtensionFilter::ExtensionFilter() :
//     Filter (module().attr("Filter")())
// {
//     object callable = 
//         make_function(&ExtensionFilter::filter_);
//    
//
//     object filter = 
//         module().attr("Filter")();
//     filter.attr("filter") = callable;
//     return Filter(filter); 
// }

}}

File: src/ackward/logging/Filter.py
from ackward import (Class,
                     Constructor,
                     Namespace,
                     TranslationUnit)

filter_doc='''\\rst
A wrapper around Python `logging.Filter
<http://docs.python.org/py3k/library/logging.html#filter-objects>`_
objects.
\\endrst'''

def tunit():
    return TranslationUnit(
        guard='INCLUDE_ACKWARD_LOGGING_FILTER_HPP',
        header_includes=[
            ('string',),
            ('ackward', 'core', 'Object.hpp'),
            ],
        impl_includes=[
            ('ackward', 'logging', 'Filter.hpp'),
            ('boost', 'python', 'import.hpp'),
            ])

def definition(env):
    t = tunit()
    ns = Namespace('ackward', 'logging', parent=t)
    cls = Class(name='Filter',
                wrapped_class='logging.Filter',
                parent=ns,
                doc=filter_doc)
    Constructor(
        parent=cls,
        doc='Create a new Filter.')

    # Constructor(
    #     signature=[('std::wstring', 'name')])

    return t
